### ISSUE TITLE
API-wrapper throttling

### ISSUE THREAD

**Issue by mingard** [dadi/publish#32]
### Issue
When errors exist in close proximity to API `fetch` requests, we have occasionally seen uncaught looping where a multitude of concurrent requests are being made to API. In development this can crash a tab, but in production it's likely to hamper API performance and potentially exceed allocated memory on Publish instance. We need to capture this with some level of throttling in Publish api-wrapper.

### TL;DR
 Add throttling to API wrapper in Publish to circumvent inevitable 💥

**Comment by eduardoboucas** [comment]
Closed in #125.
